mindsdb_native/libs/helpers/text_helpers.py

```diff
@@ -161,10 +162,20 @@ def _is_foreign_key_name(name):
     return False
 
 
+def isascii(string):
+    """
+    Used instead of str.isascii because python 3.6 doesn't have that
+    """
+    return all(ord(c) < 128 for c in string)
+
+
 def is_foreign_key(data, column_name, data_subtype, other_potential_subtypes):
     data = list(data)
 
-    foregin_key_type = DATA_SUBTYPES.INT in other_potential_subtypes or DATA_SUBTYPES.INT == data_subtype
+    if len(data) == 0:
```

Comment:
When is this needed ? The length of the data passed to this should never be == 0

Reply:
this was failing https://github.com/mindsdb/mindsdb_native/blob/stable/tests/unit_tests/libs/phases/type_deductor/test_type_deductor.py#L124

Reply:
That test is then wrong, since this should never be called on an empty column, the column should be removed before in the DataCleaner, remove the test